CONTRIBUTING.md

```diff
@@ -185,14 +189,49 @@ def do_something(value: bool = False) -> bool:
 
 Test code and branch coverage is already tracked using [coverage](https://github.com/nedbat/coveragepy) and the [pytest-cov](https://github.com/pytest-dev/pytest-cov) plugin for pytest, and it measures how much code in the `src/macaron/` folder is covered by tests.
 
-You can also run integration tests locally:
+#### Integration tests
+
+To know more about our integration test utility, please see [here](./tests/integration/README.md).
+
+We have two types of integration tests:
+1. Integration tests that are run against the Macaron Python package.
+  - Prerequisites: The dev environment must be setup first (you should only need to run once, unless you make changes to the project's dependencies).
+```bash
+make setup
+```
+  - Run these tests locally with:
 ```bash
 make integration-test
 ```
+2. Integration tests that are run against the container image.
+  - Prerequisites: Setup the integration test utility for docker.
+```bash
+make setup-integration-test-utility-for-docker
+```
+  - Run these tests locally with:
+```bash
+IMAGE=<image_name> MACARON_IMAGE_TAG=<tag> make integration-test-docker
+```
+
+Please substitute `<image_name>` and `<tag>` with the values of the container image you want to test. This container image can either be:
+- Pulled from `ghcr.io/oracle/macaron`.
+- Built from running `make build-docker`.
 
-Note that integration tests can take a long time to complete. Also the repositories that we clone for these tests will be stored under `output/` directory. If you do not remove/move this directory and run the pre-commit tool you might get errors.
+Note that integration tests can take a long time to complete.
 
 
+Each integration test case has a set of tags. Please follow these instructions on how a test case is tagged for our CI/CD pipeline:
+- If you want a test case to **only** run for the container image, use **only** `macaron-docker-image`.
+- If you want a test case to **only** run with the Macaron Python package, use **only** `macaron-python-package`.
+- To skip a test case, use `skip`. `skip` still has the same effect if it's used with other tags.
+- If you want to run a test case for both the Macaron Python package and the docker container, use `macaron-python-package` and `macaron-docker-image` tags.
```

Comment:
```suggestion
- If you want to run a test case for both the Macaron Python package and the docker container, use both `macaron-python-package` and `macaron-docker-image` tags.
```

Reply:
Addressed in https://github.com/oracle/macaron/pull/815/commits/bbc1a27658cc537a70e7f76838be55adbb9fd19b